server/src/main/java/org/elasticsearch/search/aggregations/metrics/MedianAbsoluteDeviationAggregationBuilder.java

```diff
@@ -0,0 +1,140 @@
+import org.elasticsearch.search.aggregations.AggregatorFactory;
+import org.elasticsearch.search.aggregations.support.ValueType;
+import org.elasticsearch.search.aggregations.support.ValuesSource;
+import org.elasticsearch.search.aggregations.support.ValuesSourceAggregationBuilder.LeafOnly;
+import org.elasticsearch.search.aggregations.support.ValuesSourceAggregatorFactory;
+import org.elasticsearch.search.aggregations.support.ValuesSourceConfig;
+import org.elasticsearch.search.aggregations.support.ValuesSourceParserHelper;
+import org.elasticsearch.search.aggregations.support.ValuesSourceType;
+import org.elasticsearch.search.internal.SearchContext;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+public class MedianAbsoluteDeviationAggregationBuilder extends LeafOnly<ValuesSource.Numeric, MedianAbsoluteDeviationAggregationBuilder> {
+
+    public static final String NAME = "median_absolute_deviation";
+
+    private static final ParseField COMPRESSION_FIELD = new ParseField("compression");
+
+    private static final ObjectParser<MedianAbsoluteDeviationAggregationBuilder, Void> PARSER;
+
+    static {
+        PARSER = new ObjectParser<>(NAME);
+        ValuesSourceParserHelper.declareNumericFields(PARSER, true, true, false);
+        PARSER.declareDouble(MedianAbsoluteDeviationAggregationBuilder::compression, COMPRESSION_FIELD);
+    }
+
+    public static MedianAbsoluteDeviationAggregationBuilder parse(String aggregationName, XContentParser parser) throws IOException {
+        return PARSER.parse(parser, new MedianAbsoluteDeviationAggregationBuilder(aggregationName), null);
+    }
+
+    private double compression = 1000d;
+
+    public MedianAbsoluteDeviationAggregationBuilder(String name) {
+        super(name, ValuesSourceType.NUMERIC, ValueType.NUMERIC);
+    }
+
+    public MedianAbsoluteDeviationAggregationBuilder(StreamInput in) throws IOException {
+        super(in, ValuesSourceType.NUMERIC, ValueType.NUMERIC);
+        compression = in.readDouble();
+    }
+
+    protected MedianAbsoluteDeviationAggregationBuilder(MedianAbsoluteDeviationAggregationBuilder clone,
+                                                        AggregatorFactories.Builder factoriesBuilder,
+                                                        Map<String, Object> metaData) {
+        super(clone, factoriesBuilder, metaData);
+        this.compression = clone.compression;
+    }
+
+    /**
+     * Returns the compression factor of the t-digest sketches used
+     */
+    public double compression() {
+        return compression;
+    }
+
+    /**
+     * Set the compression factor of the t-digest sketches used
+     */
+    public MedianAbsoluteDeviationAggregationBuilder compression(double compression) {
+        if (compression <= 0d) {
+            throw new IllegalArgumentException(
+                "[compression] must be greater than 0. Found [" + compression + "] in [" + name + "]");
```

Comment:
Tiny nit: could we use the ParseField here?  That way if we ever change the name we won't forge to also change the exception

Reply:
Good call